marquez_client/marquez.py

```diff
@@ -8,35 +20,37 @@
 
 class MarquezClient(object):
     API_PATH = "api/v1"
-    MARQUEZ_HOST_KEY = "MQZ_HOST"
-    MARQUEZ_PORT_KEY = "MQZ_PORT"
+    DEFAULT_TIMEOUT = 5
     c = None
     namespace = None
     dataset_api_client = None
     jobs_api_client = None
     namespace_api_client = None
 
-    def __init__(self):
-        try:
-            host_from_configs = os.environ['MQZ_HOST']
-            port_from_configs = os.environ['MQZ_PORT']
-        except KeyError as ke:
-            msg = ("Please provide proper env vars in context: "
-                   "MQZ_HOSTNAME and MQZ_PORT. Missing " + str(ke.args))
-            raise Exception(msg)
-
-        logging.info("Connecting to Marquez at %s:%s",
-                     host_from_configs, port_from_configs)
+    def __init__(self, host=None, port=None, timeout=None):
+        self.host = os.environ.get('MARQUEZ_HOST') or host
+        self.port = os.environ.get('MARQUEZ_PORT') or (
+            str(port) if port else port)
+        t = os.environ.get('MARQUEZ_TIMEOUT') or str(timeout)
+        self.timeout = int(t) if t.isnumeric() else self.DEFAULT_TIMEOUT
+
+        if not self.host or not self.port:
```

Comment:
Do we want to make sure that `self.timeout` is also set here?

Reply:
I think it shouldn't be mandatory to specify one. For now let's have a default timeout, and we can change later